src/Rules/Doctrine/ORM/EntityConstructorNotFinalRule.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Doctrine\ORM;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\ShouldNotHappenException;
+use function sprintf;
+
+/**
+ * @implements Rule<ClassMethod>
+ */
+class EntityConstructorNotFinalRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return ClassMethod::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if ($node->name->name !== '__construct') {
+			return [];
+		}
+
+		if (!$node->isFinal()) {
+			return [];
+		}
+
+		$classReflection = $scope->getClassReflection();
```

Comment:
You're reporting in all classes, not just entities.

Reply:
If I'm correct, this should work now only for entities. I've taken this code from `EntityNotFinalRule` for the detection, if the class is an entity

```php
if ($this->objectMetadataResolver->isTransient($classReflection->getName())) {
	return [];
}

$metadata = $this->objectMetadataResolver->getClassMetadata($classReflection->getName());
if ($metadata !== null && $metadata->isEmbeddedClass === true) {
	return [];
}
```